src/pymor/discretizers/builtin/cg.py

```diff
@@ -1067,39 +1073,62 @@ def discretize_stationary_cg(analytical_problem, diameter=None, domain_discretiz
         BoundaryL2Functional = BoundaryL2ProductFunctional
 
     Li = [DiffusionOperator(grid, boundary_info, diffusion_constant=0, name='boundary_part')]
+    if energy_product:
+        eLi = [DiffusionOperator(grid, boundary_info, dirichlet_clear_columns=True, diffusion_constant=0)]
     coefficients = [1.]
 
     # diffusion part
     if isinstance(p.diffusion, LincombFunction):
         Li += [DiffusionOperator(grid, boundary_info, diffusion_function=df, dirichlet_clear_diag=True,
                                  name=f'diffusion_{i}')
                for i, df in enumerate(p.diffusion.functions)]
+        if energy_product:
+            eLi += [DiffusionOperator(grid, boundary_info, diffusion_function=df, dirichlet_clear_diag=True,
+                                      dirichlet_clear_columns=True)
+                   for i, df in enumerate(p.diffusion.functions)]
         coefficients += list(p.diffusion.coefficients)
     elif p.diffusion is not None:
         Li += [DiffusionOperator(grid, boundary_info, diffusion_function=p.diffusion,
                                  dirichlet_clear_diag=True, name='diffusion')]
+        if energy_product:
+            eLi += [DiffusionOperator(grid, boundary_info, diffusion_function=p.diffusion,
+                                      dirichlet_clear_diag=True, dirichlet_clear_columns=True)]
         coefficients.append(1.)
 
     # advection part
     if isinstance(p.advection, LincombFunction):
         Li += [AdvectionOperator(grid, boundary_info, advection_function=af, dirichlet_clear_diag=True,
                                  name=f'advection_{i}')
                for i, af in enumerate(p.advection.functions)]
+        if energy_product:
+            eLi += [AdvectionOperator(grid, boundary_info, advection_function=af, dirichlet_clear_diag=True,
+                                      dirichlet_clear_columns=True)
+                   for i, af in enumerate(p.advection.functions)]
```

Comment:
I don't think it makes much sense to have the asymmetrical advection part in the energy product.